app/src/pages/Series.tsx

```diff
@@ -0,0 +1,94 @@
+import { Box, Container, Typography, LinearProgress } from "@mui/material";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { LINKS } from "./links";
+import { CompetitionCard } from "../components/CompetitionCard";
+import { CompetitionHeader } from "../components/CompetitionHeader";
+
+export const Series = () => {
+  const { t } = useTranslation();
+  const { seriesid } = useParams();
+
+  const [competitionData, setCompetitionData] = useState<any>({});
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const getData = async (seriesId: string) => {
+      const response = await fetch(LINKS.WCA.API.COMPETITION_SERIES + seriesId);
+      const seriesCompetitions = await response.json();
+
+      if (!response.ok) {
+        setHasError(true);
+        return;
+      }
+
+      let allData = await Promise.all(
+        seriesCompetitions.competitionIds.map(async (key: string) => {
+          const competitionData = await (
+            await fetch(LINKS.WCA.API.COMPETITION_INFO + key)
+          ).json();
+          const wcifData = await (
+            await fetch(LINKS.WCA.API.COMPETITION_INFO + key + "/wcif/public")
+          ).json();
+          return { ...competitionData, ...wcifData };
+        }),
+      );
+
+      setCompetitionData(allData);
+      setIsLoading(false);
+    };
+    getData(seriesid!);
+  }, [seriesid]);
+
+  if (hasError) {
+    return (
+      <Container maxWidth="xl" style={{ textAlign: "center" }}>
+        <Box marginTop="4rem">
+          <Typography
+            component="h1"
+            variant="h5"
+            fontWeight="bold"
+            gutterBottom
+          >
+            {t("competition.error")}
+          </Typography>
+        </Box>
+      </Container>
```

Comment:
This is used twice (unless there's some difference I'm not immediately seeing) so probably worth extracting it to its own file/component.
Maybe also for the loading below?